tests/store/tracking/test_databricks_rest_store.py

```diff
@@ -1469,3 +1470,339 @@ def test_unlink_traces_from_run_with_empty_list_does_nothing():
+                    "create_time": "2025-11-28T21:30:53.195Z",
+                    "last_update_time": "2025-11-28T21:30:53.195Z",
+                },
+                {
+                    "dataset_id": "dataset_2",
+                    "name": "test_dataset_2",
+                    "digest": "def456",
+                    "create_time": "2025-11-28T21:30:53.195Z",
+                    "last_update_time": "2025-11-28T21:30:53.195Z",
+                },
+            ],
+            "next_page_token": "token1",
+        },
+        {"datasets": [], "next_page_token": "token2"},
+        {
+            "datasets": [
+                {
+                    "dataset_id": f"dataset_{i}",
+                    "name": f"test_dataset_{i}",
+                    "digest": f"hash{i}",
+                    "create_time": "2025-11-28T21:30:53.195Z",
+                    "last_update_time": "2025-11-28T21:30:53.195Z",
+                }
+                for i in range(3, 11)
+            ],
+            "next_page_token": "token3",
+        },
+    ]
+
+    with (
+        mock.patch(
+            "mlflow.store.tracking.databricks_rest_store.http_request",
+            side_effect=[mock.Mock(json=lambda r=r: r) for r in responses],
+        ) as mock_http,
+        mock.patch("mlflow.store.tracking.databricks_rest_store.verify_rest_response"),
+    ):
+        result = store.search_datasets(experiment_ids=["exp_1"], max_results=5)
+
+        assert mock_http.call_count == 3
+        assert {d.name for d in result} == {
+            "test_dataset_1",
+            "test_dataset_2",
+            "test_dataset_3",
+            "test_dataset_4",
+            "test_dataset_5",
+        }
+
+
+def test_search_datasets_resume_from_composite_token():
+    creds = MlflowHostCreds("https://hello")
+    store = DatabricksTracingRestStore(lambda: creds)
+
+    response_data = {
+        "datasets": [
+            {
+                "dataset_id": f"dataset_{i}",
+                "name": f"test_dataset_{i}",
+                "digest": f"hash{i}",
+                "create_time": "2025-11-28T21:30:53.195Z",
+                "last_update_time": "2025-11-28T21:30:53.195Z",
+            }
+            for i in range(1, 16)
+        ],
+        "next_page_token": "backend_token_B",
+    }
+
+    composite_token = CompositeToken(backend_token="backend_token_A", offset=5).encode()
+
+    with (
+        mock.patch(
+            "mlflow.store.tracking.databricks_rest_store.http_request",
+            return_value=mock.Mock(json=lambda: response_data),
+        ),
+        mock.patch("mlflow.store.tracking.databricks_rest_store.verify_rest_response"),
+    ):
+        result = store.search_datasets(
+            experiment_ids=["exp_1"], max_results=10, page_token=composite_token
+        )
+
+        assert {d.name for d in result} == {f"test_dataset_{i}" for i in range(6, 16)}
+
+
+def test_search_datasets_exact_match_returns_plain_token():
```

Comment:
super nit: Why's this called a plain token? search_datasets still returns a composite token for exact match, just with offset of 0.